lib/ecto/validators.ex

```diff
@@ -24,6 +24,24 @@ defmodule PhoenixApiToolkit.Ecto.Validators do
 
   @order_by_format ~r/^(asc|desc|asc_nulls_last|desc_nulls_last|asc_nulls_first|desc_nulls_first):(\w{1,20})$/
 
+  @file_type_signatures %{
+    "image/png" => [
+      "89504E470D0A1A0A"
+    ],
+    "image/jpeg" => [
+      "FFD8FFDB",
+      "FFD8FFEE",
+      "FFD8FFE000104A4649460001",
+      "FFD8FFE1"
+    ],
+    "audio/mpeg" => [
+      "FFFB",
+      "FFF3",
+      "FFF2",
+      "494433"
```

Comment:
I'm doubting if this is a good idea. These file signatures are not an official standard, the most reliable source seems to be a wikipedia list - but it's the best way we have to check a file type apart from parsing the entire file. What happens if it turns out that there JPEGs with another signature out there? We would have to do a new release of the lib, which at one hand solves the problem for multiple dependent applications, but on the other hand it just feels wrong to make this instable list a part of a library in the first place.

Reply:
@TimPelgrim what do you think?

Reply:
I'm with @juulSme on this one. If the mime-type does not correspond with the actual file, this seems like an error on the users end. If we would actually need to know the internals of the file I would use a (maintained) extension for this. These number (besides the FFD8 of JPEG) are not something that is specified, but are a by-product of the encoding.

Another way to look at this is, it is not really necessary for use to do this, it is quite ok from an API standpoint if a file is malformed. (What if I would upload a JPEG with the form `FF D8 FF DB 00 00 00 00`, it is still a corrupt file. so this check will never be fool proof without implementing the JPEG decompressor)

Reply:
I agree that this is a very basic check, but that's exactly what this function was meant to do: a quick check to filter most garbage from being uploaded. It should not be relied on to make sure only 100% valid files are being uploaded.

I included the list in the library to prevent these "magic headers" from being defined in multiple files. I made sure that it's still possible to use a custom map. So the user can either pass their own mappings, or extend these mappings like so:

```elixir
mime_types("image/png")
|> Map.put(Base.decode16!("ABCD"), "image/png")
```

But I agree that we can't guarantee that the included list is complete. For this reason I wasn't absolutely sure if I should include this. So if you think that not including any list is better than a possibly incomplete list, I'll remove it!